src/azure-cli/azure/cli/command_modules/appconfig/_utils.py

```diff
@@ -6,41 +6,51 @@
 from knack.util import CLIError
+from azure.appconfiguration import AzureAppConfigurationClient
+from azure.mgmt.appconfiguration.models import ErrorException
 
 from ._client_factory import cf_configstore
+from ._constants import HttpHeaders
+
+logger = get_logger(__name__)
 
 
 def construct_connection_string(cmd, config_store_name):
     connection_string_template = 'Endpoint={};Id={};Secret={}'
+    # If the logged in user/Service Principal does not have 'Reader' or 'Contributor' role
+    # assigned for the requested AppConfig, resolve_resource_group will raise CLI error
+    resource_group_name, endpoint = resolve_resource_group(cmd, config_store_name)
 
     try:
-        resource_group_name, endpoint = resolve_resource_group(
-            cmd, config_store_name)
         config_store_client = cf_configstore(cmd.cli_ctx)
         access_keys = config_store_client.list_keys(
             resource_group_name, config_store_name)
         for entry in access_keys:
             if not entry.read_only:
                 return connection_string_template.format(endpoint, entry.id, entry.value)
-    except Exception:
+    except Exception as ex:
         raise CLIError(
-            'Cannot find the App Configuration {}. Check if it exists in the subscription that logged in. '.format(config_store_name))
+            'Failed to get access keys for the App Configuration "{}". Make sure that the account that logged in has "Contributor" role assigned for this App Configuration store.\n{}'.format(config_store_name, str(ex)))
 
     raise CLIError('Cannot find a read write access key for the App Configuration {}'.format(
         config_store_name))
 
 
 def resolve_resource_group(cmd, config_store_name):
-    config_store_client = cf_configstore(cmd.cli_ctx)
-    all_stores = config_store_client.list()
-    for store in all_stores:
-        if store.name.lower() == config_store_name.lower():
-            # Id has a fixed structure /subscriptions/subscriptionName/resourceGroups/groupName/providers/providerName/configurationStores/storeName"
-            return store.id.split('/')[4], store.endpoint
-    raise CLIError(
-        "App Configuration store: {} does not exist".format(config_store_name))
+    try:
+        config_store_client = cf_configstore(cmd.cli_ctx)
+        all_stores = config_store_client.list()
+        for store in all_stores:
+            if store.name.lower() == config_store_name.lower():
+                # Id has a fixed structure /subscriptions/subscriptionName/resourceGroups/groupName/providers/providerName/configurationStores/storeName"
+                return store.id.split('/')[4], store.endpoint
+    except ErrorException as ex:
+        raise CLIError("Failed to get the list of App Configuration stores for the current user. Make sure that the account that logged in has 'Reader' or 'Contributor' role assigned for the required App Configuration store.\n{}".format(str(ex)))
```

Comment:
> 'Reader' or 'Contributor' [](start = 142, length = 26)

"Make sure that the account that logged in has proper role" We can't list all possible roles and we mentioned it failed to list stores. So it indicates that cusomters need to have a role which have resources read access.

Reply:
Done